src/dtemplate.d

```diff
@@ -1774,11 +1774,28 @@ extern (C++) final class TemplateDeclaration : ScopeDsymbol
                                 Expression e;
                                 Type t;
                                 Dsymbol s;
-                                taa.index.resolve(instLoc, sc, &e, &t, &s);
+
+                                uint errors = global.startGagging();
+                                Scope *sco;
+
+                                /* The parameter isn't part of the template
+                                 * ones, let's try to find it in the
+                                 * instantiation scope 'sc' and the one
+                                 * belonging to the template itself. */
+                                taa.index.resolve(instLoc, (sco = sc), &e, &t, &s);
```

Comment:
I'd rather the `sco = sc` was broken out as a separate statement. Same for `sco = paramscope` below.